ttt.c

```diff
@@ -443,9 +443,15 @@ int main()
         } else {
             draw_board(table);
             int move;
-            do {
+            while (1) {
                 move = get_input(turn);
-            } while (table[move] != ' ');
+                if (table[move] == ' ') {
+                    break;
+                }
+                printf(
+                    "Invalid operation: O/X already placed at this "
```

Comment:
Improve this message. Check https://en.wikipedia.org/wiki/Tic-tac-toe for proper rule description.